pgbouncer/datadog_checks/pgbouncer/pgbouncer.py

```diff
@@ -173,65 +134,37 @@ def _get_connection(self, key, host='', port='', user='', password='', database_
             raise
 
         except Exception:
-            redacted_url = self._get_redacted_dsn(host, port, user, database_url)
+            redacted_url = self._get_redacted_dsn()
             message = u'Cannot establish connection to {}'.format(redacted_url)
 
             self.service_check(
-                self.SERVICE_CHECK_NAME,
-                AgentCheck.CRITICAL,
-                tags=self._get_service_checks_tags(host, port, database_url, tags),
-                message=message,
+                self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self._get_service_checks_tags(), message=message,
             )
             raise
 
-        self.dbs[key] = connection
+        self.connection = connection
         return connection
 
-    def _get_redacted_dsn(self, host, port, user, database_url):
-        if not database_url:
-            return u'pgbouncer://%s:******@%s:%s/%s' % (user, host, port, self.DB_NAME)
+    def _get_redacted_dsn(self):
+        if not self.database_url:
+            return u'pgbouncer://%s:******@%s:%s/%s' % (self.user, self.host, self.port, self.DB_NAME)
 
-        parsed_url = urlparse(database_url)
+        parsed_url = urlparse(self.database_url)
         if parsed_url.password:
-            return database_url.replace(parsed_url.password, '******')
-        return database_url
+            return self.database_url.replace(parsed_url.password, '******')
+        return self.database_url
 
     def check(self, instance):
-        host = instance.get('host', '')
-        port = instance.get('port', '')
-        user = instance.get('username', '')
-        password = instance.get('password', '')
-        tags = instance.get('tags', [])
-        database_url = instance.get('database_url')
-        use_cached = is_affirmative(instance.get('use_cached', True))
-
-        if database_url:
-            key = database_url
-        else:
-            key = '%s:%s' % (host, port)
-
-        if tags is None:
-            tags = []
-        else:
-            tags = list(set(tags))
-
         try:
-            db = self._get_connection(
-                key, host, port, user, password, tags=tags, database_url=database_url, use_cached=use_cached
-            )
-            self._collect_stats(db, tags)
+            db = self._get_connection()
+            self._collect_stats(db, self.tags)
```

Comment:
```suggestion
            self._collect_stats(db)
```
Can we fetch `self.tags` inside the _collect_stats method?

Reply:
missed that one, thanks